### ISSUE TITLE
Request that SQLFluff update `line_pos` after fixes have been applied.

### ISSUE THREAD

**Issue by Varriount** [sqlfluff/sqlfluff#1230]
I am currently developing a plugin with a rule to align the `AS` keywords found in column aliases. As part of the "fix" logic for the rule, I add a WhitespaceSegment, anchored to the "alias expression" segment. What has surprised me is that, when SQLFluff re-evaluates the modified SQL, it doesn't update the line position of the "alias expression" segment, which causes the rule to generate a new fix.

I know of ways to get around this (though I do not know if there is an intended "correct" way), however this behavior is unexpected, and at the very least should be documented.

This may be related to issue #946.

## Expected Behaviour
The code:
```python
fix = LintResult(
    anchor=alias_segment,
    fixes=[
        LintFix(
            "create",
            anchor=alias_segment,
            edit=whitespace_segment,
        )
    ],
)
```
would cause `alias_segment`'s `line_pos` attribute to be updated.

## Observed Behaviour

`alias_segment`'s `line_pos` remains unchanged between calls to the rule.

## Version
Python 3.8
SQLFluff 0.6.1

**Comment by barrywhart** [comment]
@alanmcruickshank: Is this expected or unexpected behavior?

**Comment by barrywhart** [comment]
Based on my experience, this is the expected behavior. Positions are not updated after initial parse. They are mainly useful for debugging and for messages to the user. Lint rules determine the position of elements by traversing the segment they are passed or other optional parameters such as `raw_stack`. This does make for extra effort when writing rules.

**Comment by alanmcruickshank** [comment]
I'm pretty sure this was fixed in #1027. I've been relying on this being true in some of my more recent work, and you can see that positions are updated in the `BaseSegment._position_segments()` method.

Sorry for only closing this issue now, I think it must have got missed when that PR was merged.
